### ISSUE TITLE
原版EraseNet训练问题

### ISSUE THREAD

**Issue by mr-lz** [zdyshine/Baidu-netdisk-AI-Image-processing-Challenge-handwriting#5]
作者您好，请教一下原版的EraseNet训练有什么trick吗？
我按照原版的默认配置训练，psnr只有20左右，而看您训练的能够达到28 (⊙ˍ⊙)

**Comment by zdyshine** [comment]
> 作者您好，请教一下原版的EraseNet训练有什么trick吗？ 我按照原版的默认配置训练，psnr只有20左右，而看您训练的能够达到28 (⊙ˍ⊙)

20是提交到线上的指标吗，还是线下自己划分验证集的指标？

**Comment by mr-lz** [comment]
> > 作者您好，请教一下原版的EraseNet训练有什么trick吗？ 我按照原版的默认配置训练，psnr只有20左右，而看您训练的能够达到28 (⊙ˍ⊙)
> 
> 20是提交到线上的指标吗，还是线下自己划分验证集的指标？

1081张样本，1000张用于训练，20是剩余的81张样本测的数值。

**Comment by zdyshine** [comment]
> > > 作者您好，请教一下原版的EraseNet训练有什么trick吗？ 我按照原版的默认配置训练，psnr只有20左右，而看您训练的能够达到28 (⊙ˍ⊙)
> > 
> > 
> > 20是提交到线上的指标吗，还是线下自己划分验证集的指标？
> 
> 1081张样本，1000张用于训练，20是剩余的81张样本测的数值。

那个28的PSNR应该是提交的A榜指标。你做训练实际使用的话，可以看看视觉效果。或者你用其他涨分的策略训练，看看在你划分的数据上指标提升情况。

**Comment by mr-lz** [comment]
实际视觉效果也是挺差的，所以想问下作者是不是用原版默认配置训练的呢；
再请问下两阶段的训练第一阶段的dice loss+l1 loss相比第二阶段单独的 l1 loss，是否只是将mask_loss换成了dice loss呢？

**Comment by zdyshine** [comment]
> dice loss+l1 loss

代码没问题，你可以试试直接用dice loss+l1 loss训练看看。另外单独L1 Loss时，就只用了L1， 不使用其他Loss

**Comment by mr-lz** [comment]
> > dice loss+l1 loss
> 
> 代码没问题，你可以试试直接用dice loss+l1 loss训练看看。另外单独L1 Loss时，就只用了L1， 不使用其他Loss

好的，我确认下：
dice loss + l1 loss是这个吗：
```
holeLoss =  self.l1((1 - mask) * output, (1 - mask) * gt)
validAreaLoss = self.l1(mask * output, mask * gt)
mask_loss = dice_loss(mm, 1-mask)
```

单独的L1 loss是这个吗：
```
holeLoss =  self.l1((1 - mask) * output, (1 - mask) * gt)
validAreaLoss = self.l1(mask * output, mask * gt)  
# mask_loss = dice_loss(mm, 1-mask)
```
